src/cpp/src/text2image/stable_diffusion_xl_pipeline.hpp

```diff
@@ -0,0 +1,345 @@
+                
+                std::memcpy(encoder_hidden_states_data + step, ehs_1_data + offset_1, ehs_1_shape[2] * sizeof(float));
+                std::memcpy(encoder_hidden_states_data + step + ehs_1_shape[2], ehs_2_data + offset_2, ehs_2_shape[2] * sizeof(float));
+            }
+        }
+
+        // replicate encoder hidden state to UNet model
+        if (generation_config.num_images_per_prompt == 1) {
+            // reuse output of text encoder directly w/o extra memory copy
+            m_unet->set_hidden_states("encoder_hidden_states", encoder_hidden_states);
+            m_unet->set_hidden_states("text_embeds", add_text_embeds);
+            m_unet->set_hidden_states("time_ids", add_time_ids);
+
+        } else {
+            ov::Shape enc_shape = encoder_hidden_states.get_shape();
+            enc_shape[0] *= generation_config.num_images_per_prompt;
+
+            ov::Tensor encoder_hidden_states_repeated(encoder_hidden_states.get_element_type(), enc_shape);
+            for (size_t n = 0; n < generation_config.num_images_per_prompt; ++n) {
+                batch_copy(encoder_hidden_states, encoder_hidden_states_repeated, 0, n);
+                if (batch_size_multiplier > 1) {
+                    batch_copy(encoder_hidden_states, encoder_hidden_states_repeated,
+                        1, generation_config.num_images_per_prompt + n);
+                }
+            }
+
+            m_unet->set_hidden_states("encoder_hidden_states", encoder_hidden_states_repeated);
+
+            ov::Shape t_emb_shape = add_text_embeds.get_shape();
+            t_emb_shape[0] *= generation_config.num_images_per_prompt;
+
+            ov::Tensor add_text_embeds_repeated(add_text_embeds.get_element_type(), t_emb_shape);
+            for (size_t n = 0; n < generation_config.num_images_per_prompt; ++n) {
+                batch_copy(add_text_embeds, add_text_embeds_repeated, 0, n);
+                if (batch_size_multiplier > 1) {
+                    batch_copy(add_text_embeds, add_text_embeds_repeated,
+                        1, generation_config.num_images_per_prompt + n);
+                }
+            }
+
+            m_unet->set_hidden_states("text_embeds", add_text_embeds_repeated);
+
+            ov::Shape t_ids_shape = add_time_ids.get_shape();
+            t_ids_shape[0] *= generation_config.num_images_per_prompt;
+            ov::Tensor add_time_ids_repeated(add_time_ids.get_element_type(), t_ids_shape);
+            for (size_t n = 0; n < generation_config.num_images_per_prompt; ++n) {
+                batch_copy(add_time_ids, add_time_ids_repeated, 0, n);
+                if (batch_size_multiplier > 1) {
+                    batch_copy(add_time_ids, add_time_ids_repeated,
+                        1, generation_config.num_images_per_prompt + n);
+                }
+            }
+
+            m_unet->set_hidden_states("time_ids", add_time_ids_repeated);
```

Comment:
do we need to replicate `time_ids ` ? Maybe the model can broadcast them similarly to `timestep` input ?

Reply:
It doesn't work